### DISCUSSION TITLE
[General] Issue with zkSync Portal txsync Bridge Calculation

### DISCUSSION THREAD

**Discussion by joaolago1113** [zkSync-Community-Hub/zksync-developers#259] [General] [answered]
### Environment

Mainnet

### Provide a brief description of the functionality you're trying to implement.

I am working with the zkSync portal txsync bridge and encountered an issue with the balance calculation after withdrawal transactions. 

### What is the specific issue or error you're encountering?

The portal txsync bridge of zkSync subtracts the amount that has been withdrawn from zkSync from the total amount present on Ethereum. For example, if I withdraw 0.5 ETH from zkSync, and then attempt to deposit 0.8 ETH back to zkSync, the total amount displayed for deposit is only 0.2 ETH. This seems to be an incorrect calculation of the available deposit amount.

### Can you share the error messages or logs you're receiving, if any?

No specific error messages received, but the deposit calculation seems to be incorrect as described above.

### Have you made any recent changes to the contract before encountering this issue?

No recent changes have been made to the contract that could have led to this issue.

### Are there any external libraries or contracts that your contract interacts with?

The issue is specifically related to the zkSync portal txsync bridge.

### Can you provide the relevant portions of your contract code where the issue is occurring?

Not applicable, as the issue is with the zkSync portal txsync bridge itself, not with my specific contract code.

### Have you tried to isolate the problem, and if so, what were the results?

Yes, I have tried different withdrawal and deposit amounts and consistently noticed that the amount withdrawn from zkSync is subtracted from my total Ethereum balance during subsequent deposits.

### What steps have you already taken to try to resolve the issue?

I'm a programmer so I am currently investigating the bridge code. However, I am seeking insights or suggestions from anyone with experience in zkSync bridge code to better understand why this issue is occurring.

### Additional Details

Looking for community support or insights on the zkSync portal txsync bridge, particularly regarding the calculation discrepancy observed during deposit transactions following a withdrawal.

**Comment by joaolago1113** [comment]
**Follow-up on zkSync Bridge Balance Issue**

Further investigation into the zkSync portal txsync bridge balance discrepancy shows environment-specific behavior. 

- In the local environment running the main branch, the balance is displayed correctly.
- In the production environment, the balance does not display as expected.

Below are the screenshots for comparison:

| Local Main Branch | Production Site |
|-------------------|-----------------|
| ![Local Main Branch Screenshot](https://github.com/zkSync-Community-Hub/zksync-developers/assets/22820692/518833d9-9f18-4b2d-a90c-dd7369fb2f5f) | ![Production Site Screenshot](https://github.com/zkSync-Community-Hub/zksync-developers/assets/22820692/122caa36-3adc-442f-ae18-1e54127c5bc4) |

The [full-redesign](https://github.com/matter-labs/dapp-portal/tree/full-redesign) branch also displays the balance correctly. Maybe there's an issue with the codebase of the production UI?

**Reply by JackHamer09** [reply]
@joaolago1113 just to be sure: portal.txsync.io/bridge shows wrong balance even after updating the page?
Also could you check if the issue resolves once you clear cache, local storage, session storage, etc.

**Reply by joaolago1113** [reply]
@JackHamer09 Absolutely, I've confirmed this issue completely. I cleared the cache, tried accessing it in incognito mode, used a different browser, and even tried on another computer. It appears to be a problem with the production code, as I'm running the UI locally from the main branch and from the fullredesign branch and both display the balance correctly.

**Reply by JackHamer09** [reply]
Thanks a lot for clarifying! This is very interesting issue. We (as Matter Labs) do not have full control over txSync, but I'll forward this issue to their team.

**Reply by FilipTxFusion** [reply, marked as answer]
Hi @joaolago1113, I am Filip from txSync team. Please feel free to join our discord and submit a ticket there, and we will do our best to address your issue!

**Reply by joaolago1113** [reply]
Hi @FilipTxFusion, thanks for the offer, but I'm managing with the local UI. I'm highlighting this issue to help improve the production environment for others.

**Comment by teschdV** [comment]
# The issue you're encountering with the zkSync portal txsync bridge seems to be related to how the bridge calculates available deposit amounts after a withdrawal transaction. 
# To troubleshoot this, you may want to review the deposit calculation logic in the bridge code, paying close attention to how withdrawal transactions are handled and how they affect subsequent deposit calculations.
# Additionally, reaching out to the zkSync community or support channels for insights or assistance could be beneficial in resolving this discrepancy.

**Reply by FilipTxFusion** [reply]
Hi @teschdV, this problem has been resolved on the txBridge side.

**Comment by powstingding** [comment]
The issue with the zkSync portal txsync bridge seems to stem from a miscalculation in deposit amounts after withdrawals. Investigating the bridge code and seeking insights from the community may help identify and resolve the discrepancy in balance calculations.

**Comment by tashuraschaefer** [comment]
It appears there may be a calculation discrepancy in the zkSync portal txsync bridge, where the amount withdrawn from zkSync is incorrectly subtracted from the total Ethereum balance during subsequent deposits. Seeking insights or assistance from the community or those familiar with zkSync bridge code to address this issue.

**Comment by msandiferv** [comment]
Seeking assistance with zkSync portal txsync bridge issue: incorrect balance calculation after withdrawals impacting deposit transactions.

**Comment by m4ccmocare** [comment]
The user is experiencing an issue with the balance calculation in the zkSync portal txsync bridge. Specifically, the amount withdrawn from zkSync is being subtracted from the total Ethereum balance during subsequent deposits, resulting in an incorrect calculation of the available deposit amount. The user has not made any recent changes to their contract, and the issue appears to be related to the bridge code itself. They are seeking insights or suggestions from the community to better understand why this issue is occurring. No repository link was provided.
